gateway/ln-gateway/src/lib.rs

```diff
@@ -1108,48 +1113,51 @@ impl Gateway {
             }
         };
 
+        let per_federation_routing_fees: HashMap<FederationId, RoutingFees> =
+            per_federation_routing_fees
+                .unwrap_or_default()
+                .into_iter()
+                .map(|(k, v)| (k, v.into()))
+                .collect();
+
         dbtx.insert_entry(&GatewayConfigurationKey, &gateway_config)
             .await;
+        self.update_federation_routing_fees(
+            &mut dbtx,
+            &gateway_config,
+            &per_federation_routing_fees,
+        )
+        .await?;
         dbtx.commit_tx().await;
 
-        self.update_federation_routing_fees(routing_fees, &gateway_config)
-            .await?;
+        if let Err(e) = Self::register_all_federations(self, &gateway_config).await {
```

Comment:
We no longer need to register all federations! One of the good parts of this change. We should only re-register the with the federation who's routing fees have changed.

Reply:
See comment above/below. We're are potentially changing all the federations fees to default.